staging/src/k8s.io/apiserver/pkg/server/options/encryptionconfig/plugins.go

```diff
@@ -59,25 +55,37 @@ var (
 type PluginEnabledFunc func(name string, config io.Reader) bool
 
 // Register registers a plugin Factory by name. This
-// is expected to happen during app startup.
+// is expected to happen during app startup. It does not allow
+// registering a plugin by the same name twice.
 func (ps *KMSPlugins) Register(name string, plugin Factory) {
+	ps.register(name, plugin, false)
+}
+
+// RegisterAllowRepeat registers a plugin Factory by name. This
+// is expected to happen during app startup. It allows registering
+// a plugin by the same name twice. Use Register function unless
+// you are sure that the plugin registration is stateless, and
+// the plugin is not being registered from multiple locations.
+// Does not overwrite the original plugin in case of repeat.
+func (ps *KMSPlugins) RegisterAllowRepeat(name string, plugin Factory) {
```

Comment:
To start, let's not make this part of the registry API, but contain it to the call site by guarding the registration with a sync.Once or something